.github/workflows/cypress-canary.yml

```diff
@@ -0,0 +1,73 @@
+name: Run e2e (canary demo)
+on:
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
+  push:
+    branches:
+      - main
+    paths:
+      - 'demos/canary/**/*.{js,jsx,ts,tsx}'
+      - 'cypress/integration/canary/**/*.{ts,js}'
+      - 'src/**/*.{ts,js}'
+jobs:
+  cypress:
+    name: Cypress
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        containers: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
```

Comment:
I'm not quite following what this line does, could you provide a little more context on what behaviour this leads to?

The reason I'm confused is that I'm not seeing `containers` as a key under the `matrix` property within the Github [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) but is instead a property under [`jobs`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontainer).

I know this could also be a variable that's referred to elsewhere in the tests similar to what we do with the `os` property in the `test.yml` file, but since it's not being referred to elsewhere in this file I'm not sure of the intention of the property.